PlainTasksDates.py

```diff
@@ -548,19 +548,19 @@ def extract_tag(self, point):
             tag under cursor (i.e. point)
         '''
         start = end = point
-        tag_pattern = r'(?<=\s)(\@[^\(\) ,\.]+)([\w\d\.\(\)\-!? :\+]*)'
-        line = self.view.line(point)
-        matches = re.finditer(tag_pattern, self.view.substr(line))
-        for match in matches:
-            m_start = line.a + match.start(1)
-            m_end   = line.a + match.end(2)
-            if m_start <= point <= m_end:
-                start = line.a + match.start(2)
-                end   = m_end
-                break
-        else:
-            match = None
-        tag = match.group(0) if match else ''
+        tag = ''
+        if 'tag.todo' in self.view.scope_name(point):
+            reg = self.view.extract_scope(point)
```

Comment:
How about this `@\w+((\((?=([^()]*))\3|[()]*\))*)` — it works with re module and seems suitable?
It behaves as your alternative version, which seems alright to me since the last pair of parenthesis is not separated from the tag, there is no whitespace so we can consider it is a part of tag.
What your thoughts are?



> Phantoms only appear after `@due` tags, right?

No, phantoms may appear anywhere, because any plugin can add any phantoms to any views; and it is not possible to fully access existing phantoms.


> Using view.find will work as long as no phantoms throw off the regex

Phantoms do not affect search in any way, afaik, only scopes boundaries.



> less good for any with short dates as you end up with both dates in the bracket,

The preview is supposed to be displayed only if the existing text will be replaced after pressing <kbd>tab</kbd>. I would imagine it has problems due to date format and improper use of dateutil.